### ISSUE TITLE
add ssh connection options to allow setting of JSch connection values

### ISSUE THREAD

**Issue by retrogradeorbit** [epiccastle/spire#148]
With the JSch implementation changed to the mwiede fork there needs to be a way of setting and modifying some of the lower level JSch connection configuration like `kex` and `PubkeyAcceptedAlgorithms` and `server_host_key` to make sure its still easy to connect to legacy machines. Something like:

```clojure
(ssh {:host-string "user@remotehost"
      :session {:server-host-key "+,ssh-rsa"
                :pubkey-accepted-alogrithms "+,ssh-rsa"}} ...)
```

or 

```clojure
(ssh {:host-string "user@remotehost"
      :session {:kex "+,diffie-hellman-group14-sha1"}} ...)
```

Values could be functions that take the default config setting and return the new one, perhaps.

```clojure
(ssh {:host-string "user@remotehost"
      :session {:kex-fn #(str % ",diffie-hellman-group14-sha1")}} ...)
```

**Comment by retrogradeorbit** [comment]
Options key is `:jsch-options`. Its hashmap can have standard keys like `:kex` that specify the config string completely. Or you can use functions like `:kex-fn` to pass a function. That function gets passed the old config value and returns the new. With this you can prefix, suffix, or otherwise process the config setting.

eg:

```shell
$ spire -e '(ssh {:hostname "localhost" :jsch-options {:server-host-key "ssh-rsa" :kex-fn #(str "diffie-hellman-group14-sha1," %)}})'
The authenticity of host 'localhost' can't be established.
RSA key fingerprint is SHA256:/tCQlmGVCXhwqJFq3h5aiEqD1UlUD9Eg5bDwd5yF52k.
Are you sure you want to continue connecting? y
nil
```

so, to connect to legacy ssh servers like embedded ssh server dropbear:

```clojure
:jsch-options {:kex-fn #(str % ",diffie-hellman-group14-sha1")}
```

to support old RSA/SHA1 sigs

```clojure
:jsch-options {:server-host-key-fn #(str % ",sha-rsa")
               :client-pubkey-fn #(str % ",sha-rsa")}
```

**Comment by retrogradeorbit** [comment]
Note: presently `-fn` keys don't work when utilising spire as a pod.

**Comment by retrogradeorbit** [comment]
fixed in 0.1.1
